objects/src/accounts/component/template/storage/mod.rs

```diff
@@ -0,0 +1,585 @@
+                    MapEntry {
+                        key: WordRepresentation::Template(
+                            StoragePlaceholder::new("foo.bar").unwrap(),
+                        ),
+                        value: WordRepresentation::Hexadecimal(digest!("0x2").into()),
+                    },
+                    MapEntry {
+                        key: WordRepresentation::Hexadecimal(digest!("0x2").into()),
+                        value: WordRepresentation::Template(
+                            StoragePlaceholder::new("bar.baz").unwrap(),
+                        ),
+                    },
+                    MapEntry {
+                        key: WordRepresentation::Hexadecimal(digest!("0x3").into()),
+                        value: WordRepresentation::Hexadecimal(digest!("0x4").into()),
+                    },
+                ],
+            },
+            StorageEntry::MultiSlot {
+                name: "multi".into(),
+                description: Some("Multi slot entry".into()),
+                slots: vec![2, 3, 4],
+                values: vec![
+                    WordRepresentation::Template(StoragePlaceholder::new("test.Template").unwrap()),
+                    WordRepresentation::Array(array),
+                    WordRepresentation::Hexadecimal(digest!("0xabcdef123abcdef123").into()),
+                ],
+            },
+            StorageEntry::Value {
+                name: "single-slot".into(),
+                description: Some("Slot with storage placeholder".into()),
+                slot: 5,
+                value: WordRepresentation::Template(
+                    StoragePlaceholder::new("single-slot-key").unwrap(),
+                ),
+            },
+        ];
+
+        let config = AccountComponentMetadata {
+            name: "Test Component".into(),
+            description: "This is a test component".into(),
+            version: Version::parse("1.0.0").unwrap(),
+            targets: BTreeSet::from([AccountType::FungibleFaucet]),
+            storage,
+        };
+
+        let toml = config.as_toml().unwrap();
+
+        let deserialized = AccountComponentMetadata::from_toml(&toml).unwrap();
+
+        assert_eq!(deserialized, config);
+    }
+
+    #[test]
+    pub fn test_toml() {
+        let toml_text = r#"
+            name = "Test Component"
+            description = "This is a test component"
+            version = "1.0.1"
+            targets = ["FungibleFaucet"]
+
+            [[storage]]
+            name = "map"
+            description = "A storage map entry"
+            slot = 0
+            values = [
+                { key = "0x1", value = ["{{value.test}}", "0x1", "0x2", "0x3"] },
+                { key = "{{key.test}}", value = "0x3" },
+                { key = "0x3", value = "0x4" }
+            ]
+
+            [[storage]]
+            name = "test-word"
+            description = "word"
+            slot = 1
+            value = "{{word.test}}" 
+        "#;
```

Comment:
Should we add a multi-slot entry to this test?

Reply:
This is tricky, but the `toml` crate (as part of the [toml specs](https://toml.io/en/v1.0.0#:~:text=Arbitrary-,64,-%2Dbit%20signed%20integers)) parses numbers into `i64`. For this reason, we need to parse them a strings. So if you changed your test to include quotes around the numbers within `values`, it works. I added that section to the test. 

I briefly looked and played around with [toml_edit](https://crates.io/crates/toml_edit) and it seemed like we could preprocess the toml if we wanted in order to work around this limitation. Of course, then our format would deviate from toml standards.

Reply:
Ah, makes sense, thanks! The error isn't very helpful here, but I also don't see an easy way to improve it unless going with something like `toml_edit`, so I'd probably leave it as-is for now.